agimus_controller/agimus_controller/ocp_base_croco.py

```diff
@@ -0,0 +1,226 @@
+                                t
+                            ].weight.w_robot_velocity,
+                        ]
+                    ),
+                )
+                runningCostModel.addCost(
+                    "ctrlRegGrav",
+                    uRegCost,
+                    self._OCPParams.WeightedTrajectoryPoints[t].weight.w_robot_effort,
+                )
+                runningCostModel.addCost(
+                    "gripperPoseRM",
+                    goalTrackingCost,
+                    self._OCPParams.WeightedTrajectoryPoints[
+                        t
+                    ].weight.w_end_effector_poses,
+                )
+                # Create Differential Action Model (DAM), i.e. continuous dynamics and cost functions
+                self._running_DAM = crocoddyl.DifferentialActionModelFreeFwdDynamics(
+                    self._state,
+                    self._actuation,
+                    runningCostModel,
+                )
+                # Create Integrated Action Model (IAM), i.e. Euler integration of continuous dynamics and cost
+                runningModel = crocoddyl.IntegratedActionModelEuler(
+                    self._running_DAM, self.dt
+                )
+                runningModel.differential.armature = OCPParamsCrocoBase.armature
+                self._runningModelList.append(runningModel)
+            else:
+                terminalCostModel = crocoddyl.CostModelSum(self._state)
+                terminalCostModel.addCost(
+                    "stateReg",
+                    xRegCost,
+                    np.concatenate(
+                        [
+                            self._OCPParams.WeightedTrajectoryPoints[
+                                t
+                            ].weight.w_robot_configuration,
+                            self._OCPParams.WeightedTrajectoryPoints[
+                                t
+                            ].weight.w_robot_velocity,
+                        ]
+                    ),
+                )
+                terminalCostModel.addCost(
+                    "gripperPose",
+                    goalTrackingCost,
+                    self._OCPParams.WeightedTrajectoryPoints[
+                        t
+                    ].weight.w_end_effector_poses,
+                )
+
+                self._terminal_DAM = crocoddyl.DifferentialActionModelFreeFwdDynamics(
+                    self._state, self._actuation, terminalCostModel
+                )
+
+                self._terminalModel = crocoddyl.IntegratedActionModelEuler(
+                    self._terminal_DAM, 0.0
+                )
+                self._terminalModel.differential.armature = OCPParamsCrocoBase.armature
+
+        problem = crocoddyl.ShootingProblem(
+            self.x0, self._runningModelList, self._terminalModel
+        )
+        # Create solver + callbacks
+        ocp = mim_solvers.SolverSQP(problem)
```

Comment:
i think we can have a flag in the ocp param maybe to indicate which solver use? but it might be too verbose, we could only use mim solver & csqp, which is a sqp solver when there's no constraints

Reply:
I would keep it simple and set the mim_solvers as the default for everyone.
If the performance are similar I would take the most generic solver and impose this one...